web/src/group_permission_settings.ts

```diff
@@ -1,38 +1,22 @@
-type GroupPermissionSetting = {
-    require_system_group: boolean;
-    allow_internet_group: boolean;
-    allow_owners_group: boolean;
-    allow_nobody_group: boolean;
-    allow_everyone_group: boolean;
-};
+import {page_params} from "./page_params";
+import type {GroupPermissionSetting} from "./types";
 
-const group_permission_config_dict = new Map<string, GroupPermissionSetting>([
-    [
-        "can_remove_subscribers_group",
-        {
-            require_system_group: true,
-            allow_internet_group: false,
-            allow_owners_group: false,
-            allow_nobody_group: false,
-            allow_everyone_group: true,
-        },
-    ],
-    [
-        "create_multiuse_invite_group",
-        {
-            require_system_group: true,
-            allow_internet_group: false,
-            allow_owners_group: false,
-            allow_nobody_group: true,
-            allow_everyone_group: false,
-        },
-    ],
-]);
+const realm_permission_settings = new Set(["create_multiuse_invite_group"]);
+
+const stream_permission_settings = new Set(["can_remove_subscribers_group"]);
 
 export function get_group_permission_setting_config(
     setting_name: string,
 ): GroupPermissionSetting | undefined {
-    const permission_config_dict = group_permission_config_dict.get(setting_name);
+    const permission_settings_dict = page_params.server_supported_permission_settings;
+
+    let permission_config_dict;
+    if (realm_permission_settings.has(setting_name)) {
+        permission_config_dict = permission_settings_dict.realm[setting_name];
+    } else if (stream_permission_settings.has(setting_name)) {
+        permission_config_dict = permission_settings_dict.stream[setting_name];
+    }
```

Comment:
This if/else statement is a bit weird; I guess what we want is to pass `realm` or `group` or `stream` into this function (and thus from  `get_realm_user_groups_for_dropdown_list_widget`) so that we can just check the correct data structure.

Reply:
Updated this to use a new `setting_type` variable.